### PR TITLE
Add `custom_version` parameter

### PR DESCRIPTION
**Pull request by com30n**
While using the workflow I found out that an alfred workflow version is not the same as tag in github release.
So, the changes I made can help to fix this :)

### CODE REVIEW

**Comment by mikybars** [comment]
Thanks for contributing to this repo @com30n!
I like your idea of being able to customize the version of the output workflow file with the git tag (or any string actually). I've got a couple of suggestions:
* Since the new version can be really anything and not just a git tag I'd change the name of the parameter from `tag_name` to `custom_version` to reflect that
* The custom version can go inside the `info.plist` as is but it needs to be sanitized to be part of the filename since it is provided by the user
* I would update the version with the custom version even if no `version` key is found inside the `info.plist` file

**Comment by com30n** [comment]
Thank you for the answer @almibarss!

Yeah, I agree with you it would be very helpful.

> The custom version can go inside the info.plist as is but it needs to be sanitized to be part of the filename since it is provided by the user

How do you think, which naming conversation we can use here?
Something like that `^\S{,15}`?

**Comment by com30n** [comment]
Updated the PR

File: extract_name.py [hunk 1]
```diff
@@ -42,6 +42,20 @@ def main():
...
+    if custom_version:
+        # if the custom version is `github.ref`, then sanitize it to exact tag name
+        custom_version = custom_version.replace('refs/tags/', '')
+        sanitized_custom_version = custom_sanitize_regexp.findall(custom_version)[0]
+        if sanitized_custom_version:
+            info['version'] = custom_version
+            plistlib.writePlist(info, info_plist)
+
     if 'version' in info:
         workflow_file += '-{}'.format(info['version'])
```
Comment (mikybars) [inline extract_name.py]: Actually you don't need to extract the tag name as `github.ref_name` already contains the information that you want as opposed to `github.ref`:

image

https://docs.github.com/en/actions/learn-github-actions/contexts

```suggestion

    custom_version = os.environ.get("CUSTOM_VERSION")
    if custom_version:
        info['version'] = custom_version
        plistlib.writePlist(info, info_plist)

    if 'version' in info:
        workflow_file += '-{}'.format(safename(info['version']))
```

**Review by mikybars — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ This action needs a file named `info.plist` with the metadata of your workflow i
 
 * `workflow_dir`: Directory containing the sources of the workflow (defaults to `workflow`)
 * `exclude_patterns`: List of excluded files/directories
+* `custom_version`: Custom version of your alfred workflow (like git tag name). It will sanitize the version using the regexp `^\S{,15}`. It updates the workflow version (it updates `info.plist` version)
```
Comment (mikybars) [inline README.md]:
```suggestion
* `custom_version`: String that will override the workflow version (a git tag name for example).
```

File: action.yml [hunk 3]
```diff
@@ -13,12 +13,17 @@ inputs:
     description: 'List of excluded files/directories'
     required: false
     default: ''
+  custom_version:
+    description: 'The name of the tag. This should come from the webhook payload, `github.GITHUB_REF` when a user pushes a new tag'
```
Comment (mikybars) [inline action.yml]:
```suggestion
    description: 'String that will override the workflow version'
```

**Review by mikybars — CHANGES_REQUESTED** [review]

**Comment by com30n** [comment]
Great, thank you @almibarss  for the fixes!

**Comment by mikybars** [comment]
lgtm 👍🏾 Thank you @com30n

**Pull request merged**